pints/tests/test_residuals_diagnostics.py

```diff
@@ -183,6 +183,44 @@ def test_plot_residuals_autocorrelation(self):
             posterior_interval=1.5
         )
 
+    def test_plot_residuals_vs_output(self):
+        # Test the function which plots residuals against output magnitudes
+
+        # Test that it runs with an optimisation result
+        pints.residuals_diagnostics.plot_residuals_vs_output(
+            np.array([self.found_parameters1]),
+            self.problem1
+        )
+
+        # Test that it runs with multiple ouputs and MCMC
+        fig = pints.residuals_diagnostics.plot_residuals_vs_output(
+            self.samples2[0],
+            self.problem2
+        )
+
+        # Test that the multiple output figure has multiple axes
+        self.assertGreaterEqual(len(fig.axes), 2)
+
+        # Check the message when the input is wrong dimension
+        self.assertRaisesRegexp(
+            ValueError,
+            r'\`parameters\` must be of shape \(n_samples\, n_parameters\)\.',
```

Comment:
You don't have to stick the whole message in here. Just enough so that we know that it's definitely the error we wanted to check for. (The exact wording isn't important, the fact that we check some of the wording is a bit of a side-effect)